servo/connectors/kubernetes.py

```diff
@@ -2003,6 +2016,578 @@ async def get_restart_count(self) -> int:
+def default_kubernetes_json_serializer(o: Any) -> Any:
+    if isinstance(o, (datetime.datetime, datetime.date)):
+        return o.isoformat()
+    raise TypeError(f'Object of type {o.__class__.__name__} '
+                        f'is not JSON serializable')
+
+# https://github.com/kubernetes-client/python/issues/977#issuecomment-592030030
+class FakeKubeResponse:
+    """Mocks the RESTResponse object as a workaround for kubernetes python api_client deserialization"""
+    def __init__(self, obj):
+        self.data = json.dumps(obj, default=default_kubernetes_json_serializer)
+
+# Use alias generator so that lower camel case can be parsed to snake case properties to match k8s python client behaviour
+def to_lower_camel(string: str) -> str:
+    split = string.split('_')
+    return split[0] + ''.join(word.capitalize() for word in split[1:])
+
+class RolloutBaseModel(pydantic.BaseModel):
+    class Config:
+        # arbitrary_types_allowed = True
+        alias_generator = to_lower_camel
+        allow_population_by_field_name = True
+
+# Pydantic type models for argo rollout spec: https://argoproj.github.io/argo-rollouts/features/specification/
+# https://github.com/argoproj/argo-rollouts/blob/master/manifests/crds/rollout-crd.yaml
+# NOTE/TODO: fields typed with Any should maintain the same form when dumped as when they are parsed. Should the need
+#   arise to interact with such fields, they will need to have an explicit type defined so the alias_generator is applied
+class RolloutV1LabelSelector(RolloutBaseModel): # must type out k8s models as well to allow parse_obj to work
+    match_expressions: Any
+    match_labels: Optional[Dict[str, str]]
+
+class RolloutV1ObjectMeta(RolloutBaseModel):
+    annotations: Optional[Dict[str, str]]
+    cluster_name: Optional[str]
+    creation_timestamp: Optional[datetime.datetime]
+    deletion_grace_period_seconds: Optional[int]
+    deletion_timestamp: Optional[datetime.datetime]
+    finalizers: Optional[List[str]]
+    generate_name: Optional[str]
+    generation: Optional[int]
+    labels: Optional[Dict[str, str]]
+    managed_fields: Any
+    name: Optional[str]
+    namespace: Optional[str]
+    owner_references: Any
+    resource_version: Optional[str]
+    self_link: Optional[str]
+    uid: Optional[str]
+
+class RolloutV1EnvVar(RolloutBaseModel):
+    name: str
+    value: Optional[str]
+    value_from: Any
+
+class RolloutV1ContainerPort(RolloutBaseModel):
+    container_port: int
+    host_ip: Optional[str]
+    host_port: Optional[int]
+    name: Optional[str]
+    protocol: Optional[str]
+
+class RolloutV1ResourceRequirements(RolloutBaseModel):
+    limits: Optional[Dict[str, str]]
+    requests: Optional[Dict[str, str]]
+
+class RolloutV1Container(RolloutBaseModel):
+    args: Optional[List[str]]
+    command: Optional[List[str]]
+    env: Optional[List[RolloutV1EnvVar]]
+    env_from: Any
+    image: str
+    image_pull_policy: Optional[str]
+    lifecycle: Any
```

Comment:
Why bother to even have these fields if you aren't going to type them? We can just use the `Config` to ignore unspecified fields if you don't want to build them out but specifying fields as `Any` isn't helpful and effectively backslides into turning the data model into a glorified dictionary

Reply:
We have no need to interact with these fields in the context of the connector but can't exclude them from the model because they would not be included in copying the rollout template spec into a standalone pod. By leaving the type as Any, we can effectively say we don't care about the typing of these fields because they came from an already validated Kubernetes object and will only ever be used to populate a new Kubernetes object.

I don't see a problem with the model being a glorified dictionary in this context since we are only interested in passing through the data from reads to any write operations and these models are meant to be temporary anyway until the generated models can be implemented.

LMK If you still disagree with the justification above, and I can expand the models to include fields we currently specify as Any.